### ISSUE TITLE
Hold Fire Against Capture-Marked Enemies

### ISSUE THREAD

**Issue by Brian-Catcow-B** [beyond-all-reason/Beyond-All-Reason#3156]
### Description

Have each enemy structure keep track of whether or not it has been queued for capture by a given player so that that player's units do not attack it. 

### Why?

The Capture mechanic is a very cool part of BAR, but manually setting "Hold Fire" on the commander just to capture a metal extractor without taking it to half hp seems unnecessary. There are also situations where capturing enemy units would be a lot more viable if your own units would immediately stop firing at the unit that has been queued for capture. 

### What area of the game does this feature request pertain?

Gameplay

### Other (optional)

It is very important to not end up with a "poisoned state" where an enemy unit is marked for capture but is not actually in a capture queue. One way to solve this is to keep track which unit has the capture-marked unit in its queue and then each time the capture-marked unit would be targeted, ensure that the unit attempting to capture it is indeed still trying to capture it. Of course, with giving commanders to teammates and with decoy commanders being able to capture, this would have to be a list of capturers. 

Also, this probably shouldn't propogate through to teammates because of easy trolling.

**Comment by PtaQQ** [comment]
Already planned in the design doc :) Lets leave the issue open for more visibility, perhaps someone picks it up.

**Comment by sprEEEzy** [comment]
Return fire would be better.

**Comment by SamWain** [comment]
I'm happy to work on implementing this

**Comment by WatchTheFort** [comment]
> I'm happy to work on implementing this

We look forward to the pull request 😸

**Comment by SamWain** [comment]
For clarification, is the intent to stop ALL allyTeam units from auto targeting active capture targets?

**Comment by WatchTheFort** [comment]
> For clarification, is the intent to stop ALL allyTeam units from auto targeting active capture targets?

No, only your own, a player's units should never obey somebody else's orders.

**Comment by SamWain** [comment]
Understood

**Comment by ben-everson** [comment]
Has this issue been resolved?

**Comment by SamWain** [comment]
> Has this issue been resolved?

Not yet, ended up being more complicated than it seemed at first, working on a engine PR to make this fix possible.

**Comment by SamWain** [comment]
Have a draft pr for this issue ready for it's first review, here #4740
